### PR TITLE
Replace prints with logging

### PR DESCRIPTION
**Pull request by simleo**
Replaces prints with logging statements with a configurable level.

```
[simleo:scripts] (cleanup) % python3 create_dpkg.py ../examples/TrackMate/example_2/FakeTracks.xml                                 
2017-05-22 15:08:51,911 INFO: [create_dpkg] Trying default config file location: "/Users/simleo/git/biotracks/examples/TrackMate/example_2/biotracks.ini"
2017-05-22 15:08:51,912 INFO: [TrackMateReader] TrackMateReader Reading "../examples/TrackMate/example_2/FakeTracks.xml"
2017-05-22 15:08:51,915 INFO: [TrackMateReader] Reading a TrackMate XML file version 3.4.2
2017-05-22 15:08:52,073 INFO: [create_dpkg] tabular files written to "dp"
2017-05-22 15:08:53,099 INFO: [create_dpkg] json file written to "dp"
2017-05-22 15:08:56,503 INFO: [create_dpkg] normalizing dataset to the origin of the coordinate system...
2017-05-22 15:08:57,460 INFO: [create_dpkg] computing turning angles...
[simleo:scripts] (cleanup) % python3 create_dpkg.py ../examples/TrackMate/example_2/FakeTracks.xml --log-level=DEBUG
2017-05-22 15:09:37,952 INFO: [create_dpkg] Trying default config file location: "/Users/simleo/git/biotracks/examples/TrackMate/example_2/biotracks.ini"
2017-05-22 15:09:37,953 INFO: [TrackMateReader] TrackMateReader Reading "../examples/TrackMate/example_2/FakeTracks.xml"
2017-05-22 15:09:37,956 INFO: [TrackMateReader] Reading a TrackMate XML file version 3.4.2
2017-05-22 15:09:37,956 DEBUG: [TrackMateReader] Found 133 unique spots
2017-05-22 15:09:37,958 DEBUG: [TrackMateReader] Found 128 unique edges
2017-05-22 15:09:38,007 DEBUG: [TrackMateReader] Created 6 links
2017-05-22 15:09:38,121 INFO: [create_dpkg] tabular files written to "dp"
2017-05-22 15:09:39,031 INFO: [create_dpkg] json file written to "dp"
2017-05-22 15:09:39,086 DEBUG: [push_to_pandas] Created 5 tracks
2017-05-22 15:09:39,090 DEBUG: [create_dpkg] Datapackage pushed to pandas
2017-05-22 15:09:39,090 DEBUG: [create_dpkg] Number of rows: 133
2017-05-22 15:09:39,090 DEBUG: [create_dpkg] Number of columns: 4
2017-05-22 15:09:42,595 INFO: [create_dpkg] normalizing dataset to the origin of the coordinate system...
2017-05-22 15:09:44,353 INFO: [create_dpkg] computing turning angles...
[simleo:scripts] (cleanup) % python3 create_dpkg.py ../examples/TrackMate/example_2/FakeTracks.xml --log-level=WARNING
```

Also refactors `readfile.py` into reader objects and performs other cleanup.

### CODE REVIEW

**Comment by sbesson** [comment]
@simleo: #22 is in, can you either rebase or merge `origin/master` into this?

**Comment by simleo** [comment]
@sbesson rebased

**Comment by simleo** [comment]
Travis failure seems transient, trying with close and reopen

**Comment by simleo** [comment]
Travis happy again after cbcaddb

**Review by pcmasuzzo — COMMENTED** [review]
This is OK for me. I ran it and tested/looked at examples.

The plots for TrackMate example 1 are not reproduced, however, but this is an issues in the plotting code.
@simleo - would you like me to look at which parts of the code, so you can push one more commit before we merge this?

**Comment by simleo** [comment]
@pcmasuzzo yes, thanks. If you already know where the bug is we might as well fix it here.

**Comment by pcmasuzzo** [comment]
Yes - the plotting is called on '`TRACK_ID`' alone, while should be called on both '`TRACK_ID`' and '`LINK_ID`'.

 In '`create_dpkg.py`':

Current line 119 is: 
 `plot.plotXY(objects_links_tracks, 'TRACK_ID', x, y)`
Add (another line):
`plot.plotXY(objects_links_tracks, 'LINK_ID', x, y)`

Current line 124 is:
`plot.plotXY(norm, 'TRACK_ID', x + 'norm', y + 'norm')`
Add (another line):
`plot.plotXY(norm, 'LINK_ID',  x + 'norm', y + 'norm')`

This will produce more plots (some of them not really useful), but covers all the possibilities in the examples.

This should fix #24

**Comment by simleo** [comment]
@pcmasuzzo added in f2fd195

**Comment by pcmasuzzo** [comment]
For me this can be merged! Thanks @simleo

File: biotracks/pushtopandas.py [hunk 1]
```diff
@@ -6,33 +6,32 @@
...
+    """Push a datapackage to a pandas storage.
 
-    Keyword arguments:
-    directory -- the directory to look into for the json descriptor
-    object_id_cmso -- the join_id specified in the .ini file
+    arguments:
+    directory -- the datapackage directory containing the json file
+    object_id_cmso -- the object id
     """
+    if log_level is None:
```
Comment (sbesson) [inline biotracks/pushtopandas.py]: Does this need to be special cased here or should `log_level=None` be handled by `get_logger` ?

**Review by sbesson — COMMENTED** [review]
The usage of loggers makes complete sense. Only question is whether the `NullLogger` case should be handled internally rather than by the consumers.

I didn't have time to go through the `readfile.py` changes as the API is largely refactored in addition to the logging fixes. Generally migrating to reader classes sounds sensible but this makes the diff quite large and I could use some more time to review the design of this new API.

**Comment by simleo** [comment]
> Only question is whether the NullLogger case should be handled internally rather than by the consumers.

Nice idea, makes the code simpler. Done in f4a108d.

> Generally migrating to reader classes sounds sensible but this makes the diff quite large

I couldn't find a convenient way to decouple into shorter diffs. I've trusted the unit tests to check that everything still worked :)

**Pull request merged**